src/smart-components/role/add-role-new/add-role-wizard.js

```diff
@@ -57,6 +57,40 @@ const AddRoleWizard = ({ history: { push } }) => {
     push('/roles');
   };
 
+  const onSubmit = async (formData) => {
+    const {
+      'role-name': name,
+      'role-description': description,
+      'role-copy-name': copyName,
+      'role-copy-description': copyDescription,
+      'add-permissions-table': permissions,
+      'cost-resources': resourceDefinitions,
+      'role-type': type,
+    } = formData;
+    const roleData = {
+      applications: [formData.application],
+      description: (type === 'create' ? description : copyDescription) || '',
+      name: type === 'create' ? name : copyName,
+      access: permissions.map(({ uuid: permission }) => ({
+        permission,
+        resourceDefinitions: resourceDefinitions?.find((r) => r.permission === permission)
+          ? [
+              {
+                attributeFilter: {
+                  key: `cost-management.${permission.split(':')[1]}`,
+                  operation: 'in',
+                  value: resourceDefinitions?.find((r) => r.permission === permission).resources,
+                },
+              },
+            ]
+          : [],
+      })),
+    };
+    const role = await dispatch(createRole(roleData));
+    fetchRoles().then(push('/roles'));
+    return role;
```

Comment:
Haven't tested it in UI, but it this seems like that after a click on the submit, the form is still opened until roles are fetched, so it would be possible to click multiple times.. is that right?

Reply:
This feels a bit overcomplicated, it will create new role, waits untill that is done, starts fetching roles, fills in the role to form and once the roles are fetched it redirects to said page. We should probably wait for the role to create and if it fails show user what happens, so we'll wait for the role once that succeeds start fetching roles, don't wait for it to happen and redirect to roles. It will redirect and once that is fetched it'll show the full table.

Reply:
I think, that your concerns should be resolved in second commit.